watchfaces/007-bold-hour-bebas.qml

```diff
@@ -21,103 +23,197 @@
-        smooth: true
-        renderStrategy: Canvas.Cooperative
-        onPaint: {
-            var ctx = getContext("2d")
-            var rot1 = (0 -15 )*6 *0.01745329252
-            var rot2 = (60 -15 )*6 *0.01745329252
-            ctx.reset()
-            ctx.lineWidth = 3
-            ctx.fillStyle = Qt.rgba(0.184, 0.184, 0.184, 0.95)
-            ctx.beginPath()
-            ctx.moveTo(minuteX, minuteY)
-            ctx.arc(minuteX, minuteY, width / 8.6, rot1, rot2, false);
-            ctx.lineTo(minuteX, minuteY);
-            ctx.fill();
+        visible: nightstandMode.active
+        layer {
+            enabled: true
+            samples: 4
+            smooth: true
+            textureSize: Qt.size(nightstandMode.width * 2, nightstandMode.height * 2)
+        }
+
+        Shape {
+            id: chargeArc
+
+            property real angle: batteryChargePercentage.percent * 360 / 100
+            // radius of arc is scalefactor * height or width
+            property real arcStrokeWidth: .04
+            property real scalefactor: .5 - (arcStrokeWidth / 2)
+            property real chargecolor: Math.floor(batteryChargePercentage.percent / 33.35)
+            readonly property var colorArray: [ "red", "yellow", Qt.rgba(.318, 1, .051, .9)]
+
+            anchors.fill: parent
+            //smooth: true
+            //antialiasing: true
+
+            ShapePath {
+                fillColor: "transparent"
+                strokeColor: chargeArc.colorArray[chargeArc.chargecolor]
+                strokeWidth: parent.height * chargeArc.arcStrokeWidth
+                capStyle: ShapePath.FlatCap
+                joinStyle: ShapePath.MiterJoin
+                startX: width / 2
+                startY: height * ( .5 - chargeArc.scalefactor)
+
+                PathAngleArc {
+                    centerX: parent.width / 2
+                    centerY: parent.height / 2
+                    radiusX: chargeArc.scalefactor * parent.width
+                    radiusY: chargeArc.scalefactor * parent.height
+                    startAngle: -90
+                    sweepAngle: chargeArc.angle
+                    moveToStart: false
+                }
+            }
         }
     }
 
-    Text {
-        id: minuteDisplay
-        property var rotM: (wallClock.time.getMinutes() - 15)/60
-        property var centerX: parent.width/2-width/2
-        property var centerY: parent.height/2-height/2
-        font.pixelSize: parent.height/5.24
-        font.family: "BebasKai"
-        font.styleName:'Condensed'
-        color: "white"
-        opacity: 1.00
-        x: centerX+Math.cos(rotM * 2 * Math.PI)*parent.width*0.364
-        y: centerY+Math.sin(rotM * 2 * Math.PI)*parent.width*0.364
-        text: wallClock.time.toLocaleString(Qt.locale(), "mm")
+    MceBatteryLevel {
+        id: batteryChargePercentage
+    }
+
+    MceBatteryState {
+        id: batteryChargeState
+    }
+
+    MceCableState {
+        id: mceCableState
```

Comment:
Why are mceCableState and batteryChargeState necessary ? Isn't batteryChargePercentage the only one you need ?

Reply:
Yepp, both where used pre berosets nightstand implementation. Not necessary anymore.